### PR TITLE
feat: remove hardcode abi

### PR DESCRIPTION
**Pull request by joshualyguessennd**

### CODE REVIEW

**Comment by jaketimothy** [comment]
Task linked: [CU-8685qwybh Replace hardcoded ABIs with sbt-deployments import in examples](https://app.clickup.com/t/8685qwybh)

File: script/request_cancel.ts [hunk 1]
```diff
@@ -1,5 +1,11 @@
-import "dotenv/config";
+require('dotenv').config({ path: '../.env' });
```
Comment (jaketimothy) [inline script/request_cancel.ts:1]: Remove path repointing. The script should be executed from the base directory.

File: script/request_order.ts [hunk 2]
```diff
@@ -1,5 +1,12 @@
-import "dotenv/config";
+require('dotenv').config({ path: '../.env' });
```
Comment (jaketimothy) [inline script/request_order.ts:1]: Execute from base directory

**Review by jaketimothy — CHANGES_REQUESTED** [review]

File: script/request_order.ts (hunk 2 above)
Comment (joshualyguessennd) [reply script/request_order.ts:1]: I had issue with my environment , I wasn't able to find PRIVATE_KEY and .env  even when I load it that why I use path repointing

File: script/request_order.ts [hunk 3]
```diff
@@ -1,5 +1,12 @@
 import "dotenv/config";
 import { ethers } from "ethers";
+import fs from 'fs';
+
+const buyProcessorDataPath = './sbt-deployments/src/v0.1.0/buy_processor.json';
+const buyProcessorData = JSON.parse(fs.readFileSync(buyProcessorDataPath, 'utf8'));
+const buyProcessorAbi = buyProcessorData.abi;
+const buyProcessorAddress = "0x1754422ef9910572cCde378a9C07d717eC8D48A0";
```
Comment (jaketimothy) [inline script/request_order.ts:8]: Use sepolia deployment address from json

File: script/request_cancel.ts [hunk 4]
```diff
@@ -1,5 +1,11 @@
 import "dotenv/config";
 import { ethers } from "ethers";
+import fs from 'fs';
+
+const buyProcessorDataPath = './sbt-deployments/src/v0.1.0/buy_processor.json';
+const buyProcessorData = JSON.parse(fs.readFileSync(buyProcessorDataPath, 'utf8'));
+const buyProcessorAbi = buyProcessorData.abi;
+const buyProcessorAddress = "0x1754422ef9910572cCde378a9C07d717eC8D48A0";
```
Comment (jaketimothy) [inline script/request_cancel.ts:8]: Use sepolia deployment address from json

File: script/go/request_order.go [hunk 5]
```diff
@@ -139,7 +167,13 @@ func main() {
...
 	processorAddress := common.HexToAddress(BuyProcessorAddress)
 	fmt.Println("buy processor address at:", processorAddress)
-	buyProcessorAbi, err := abi.JSON(strings.NewReader(buyProcessorAbiString))
+	// Load ABI from file
+	abiString, err := loadABI("../sbt-deployments/src/v0.1.0/buy_processor.json")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	buyProcessorAbi, err := abi.JSON(strings.NewReader(abiString))
```
Comment (jaketimothy) [inline script/go/request_order.go:176]: Use sepolia deployment address from json for BuyProcessorAddress

**Review by jaketimothy — CHANGES_REQUESTED** [review]

File: script/request_order.ts (hunk 3 above)
Comment (jaketimothy) [reply script/request_order.ts]: Use chainid to look up address to get correct network

File: script/request_cancel.ts (hunk 4 above)
Comment (jaketimothy) [reply script/request_cancel.ts]: Use chainid to look up address to get correct network

File: script/go/request_order.go (hunk 5 above)
Comment (jaketimothy) [reply script/go/request_order.go:201]: Use chainid to look up address to get correct network

**Review by jaketimothy — APPROVED** [review]

**Comment by jaketimothy** [comment]
We will need to replace the sbt-deployments references with imports from SDK when it's ready.

**Pull request merged**